core/src/main/java/org/infinispan/xsite/irac/DefaultIracManager.java

```diff
@@ -201,7 +218,7 @@ public void onTopologyUpdate(CacheTopology oldCacheTopology, CacheTopology newCa
       if (!addedSegments.isEmpty()) {
          // request state from old primary owner
          // and group new segments by primary owner
-         Map<Address, IntSet> primarySegments = new HashMap<>();
+         Map<Address, IntSet> primarySegments = new HashMap<>(newCacheTopology.getMembers().size());
          for (int segment : addedSegments) {
             Address primary = newCacheTopology.getWriteConsistentHash().locatePrimaryOwnerForSegment(segment);
             primarySegments.computeIfAbsent(primary, DefaultIracManager::newIntSet).add(segment);
```

Comment:
Would also be better if you could supply the max segment to the newIntSet method otherwise it will have to resize a few times.